_posts/2020-02-10-waku-current-state.md

```diff
@@ -0,0 +1,141 @@
+
+For more details, see [implementation matrix](https://specs.vac.dev/waku/waku.html#appendix-b-implementation-notes).
+
+In terms of end user applications, work is currently in progress to integrate it into the [Status core app](https://github.com/status-im/status-react/pull/9949) and is expected to be released in their upcoming 1.1 release.
+
+TODO: Fact check this with Adam and Kim
+
+**Simulation:**
+
+We've got a [simulation](https://github.com/status-im/nimbus/tree/master/waku#testing-waku-protocol) in the Nimbus client that verifies - or rather, fails to falsify - the scalability model posted in an [earlier post](https://vac.dev/fixing-whisper-with-waku). More on this below.
+
+## How many users does Waku support?
+
+This is our current understanding of how many users a network running the Waku protocol can support. Specifically in the context of the Status chat app, since that's the most immediate consumer of Waku. It should generalize fairly well to most deployments, but YMMW.
+
+**tldr (for Status app):**
+- beta: 100 DAU
+- v1: 1k DAU
+- v1.1 (waku only): 10k DAU (up to x10 with deployment hotfixes)
+- v1.2 (waku+dns): 100k DAU (can optionally be folded into v1.1)
+
+*Assuming 10 concurrent users = 100 DAU. Estimate uncertainty increases for each order of magnitude until real-world data is observed.*
+
+As far as we know right now, there is one immediate bottleneck, a very likely bottleneck after that, and a third bottleneck that is still conjecture but not unlikely to appear. These are:
+
+- Bottleneck 1 - Receive bandwidth for end user clients (aka ‘Fixing Whisper with Waku’)
+- Bottleneck 2 - Nodes and cluster capacity (aka ‘DNS based node discovery’)
+- Bottleneck 3 - Full node traffic (aka ‘the routing / partition problem’)
+
+We've already seen the first bottleneck being discussed in the initial post. Dean wrote a post on [DNS based discovery](https://vac.dev/dns-based-discovery) which explains how we will address the likely second bottleneck. More on the third one in future posts.
+
+For more details on these bottlenecks, uncertainty and mitigations, see [Scalability estimate: How many users can Waku and the Status app support?](https://discuss.status.im/t/scalability-estimate-how-many-users-can-waku-and-the-status-app-support/1514).
+
+TODO: Elaborate on bottleneck 3, kad etc
+
+## Simulation
+
+The ultimate test is real-world usage. Until then, we have a simulation. Here's some data on this.
+
+![](assets/img/waku_simulation.jpeg)
+
+Here we outline two network topologies, Star and full mesh, with 6 randomly connected nodes and 2 X, sending 10000 random envelopes over random topics, including invalid envelopes.
+
+For light node, bloom filter is set to 1% false positive. It shows the number of valid and invalid envelopes received for the different nodes.
```

Comment:
With the settings in the quicksim the false positive probability is almost 10% (p=0.087). 
Light node and Waku light node subscribe to 100 randomly selected topics (that makes n=100, k=3, m=512 for the bloom filter). 

Next, one of the full nodes sends 1 envelope over 1 of the 100 topics that the two light nodes subscribed on. This is to test that the subscription works, else the 10k envelopes will of course not be received (come to think of it, this should be done for all 100 topics to be a proper check).
Then, that same full node sends 10000 envelopes over random topics.

This explains why you see 10001 for the full nodes, and 1 for the Waku light node (as in fact, with just 10000 envelopes to random topics, the chance is very low that one of the Waku light node topics gets "hit").

Reply:
Ah should've looked more carefully, thanks for the correction and added detail :)

Reply:
fixed